libindy/src/api/ledger.rs

```diff
@@ -1616,4 +1616,45 @@ pub extern fn indy_parse_get_revoc_reg_delta_response(command_handle: i32,
     trace!("indy_parse_get_revoc_reg_delta_response: <<< res: {:?}", res);
 
     res
-}
\ No newline at end of file
+}
+
+pub type CustomTransactionParser = extern fn(reply_from_node: *const c_char, parsed_sp: *mut *const c_char) -> ErrorCode;
+pub type CustomFree = extern fn(data: *const c_char) -> ErrorCode;
+
+#[no_mangle]
+pub extern fn indy_register_transaction_parser_for_sp(command_handle: i32,
+                                                      pool_handle: i32,
+                                                      txn_type: *const c_char,
+                                                      parser: Option<CustomTransactionParser>,
```

Comment:
I prefer to have Option inside of type definition. Otherwise user can get a feeling that this parameter is optional

Reply:
We always use Option around custom callback type. We can change our approach, but I don't like to mix different in codebase. So after additional discussion we can raise separate PR for API changes (this, payments, wallet, etc)